File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] A) {
        int sum = -(int)1e9;
        int maxSum = -(int)1e9;
        int n = A.length;
        for(int i=0; i<n; i++){
            sum= Math.max(A[i], sum+A[i]);
            if(sum >= maxSum){
                maxSum = sum;
            }
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ptr=0
        summ=0
        ans=nums[0]
        while(ptr<len(nums)):
            if summ+nums[ptr]<=0:
                summ=0
                ans=max(ans,nums[ptr])
            else:
                summ+=nums[ptr]
                ans=max(ans,summ)
            ptr+=1
        return ans
        